spec/README.md

```diff
@@ -66,3 +67,23 @@ A **_term_** looks like this when it is defined in this specification.
 A reference to a _term_ looks like this.
 
 > Examples are non-normative and styled like this.
+
+### Stability Policy
+
+Updates to this specification MUST NOT change the formatting or other behaviour
+of valid messages written for earlier versions of this specification
+that only use functions and expression attributes defined in this specification.
+Later specification versions MAY make previously invalid messages valid.
+
+Updates to this specification MUST NOT introduce message syntax that,
+when parsed according to earlier versions of this specification,
+would produce syntax or data model errors.
+Such messages MAY produce errors when formatted
+according to an earlier version of this specification.
+
+Later versions of this specification MAY define new function names,
```

Comment:
I think we want to start separating the specification and registry now, since they are likely to have different life-cycles. Perhaps s/specification/registry/ along the lines of:

```suggestion
Unicode will only register or consider for registration function names,
function option names,
or expression attribute names 
consisting of characters in the ranges a-z, A-Z, and 0-9.
All other names are reserved for the use of implementations or users.
```

... and note that this policy begins now.

Reply:
I would prefer to keep the reference to "specification", as it currently includes the "registry". As we've identified in the discussions on markup and expression attributes as well as the registry itself, we'll need to have space somewhere describing the behaviour associated with some of the specified names.

If we do end up with a spec that does not by name refer to _any_ of these but relegates that to a separate document and we have a settled name for that document such as "schema", then this could be changed.

Reply:
Notice that my suggested text doesn't mention the spec or the registry. Maybe give it a second look? Also, it gives a durable promise of what we'll consider for names.

Reply:
It does, however, use the terms "register" and "registration", which might not be what we'll want to use if we end up calling the document defining functions and other things something else than "registry".

Also, looking at it again I notice that it's binding "Unicode" and not "this specification". I think that's too large a scope, as it could be seen as a restriction on what ICU can do, or what other Unicode specifications can do.

Reply:
By using Unicode and not "this specification" I can encompass all aspects of prescribed naming, regardless of form: we are promising not to use different terms in MF2. I agree that we need to narrow it to MF2 and not prohibit ICU from making an implementation choice (especially if that choice were to create a local-to-ICU definition consistent with the beat practice found later in this section).

Perhaps then:

```suggestion
From version 2.0, MessageFormat will only reserve, define, or register 
function names,
function option names,
or expression attribute names 
consisting of characters in the ranges a-z, A-Z, and 0-9.
All other names in these categories are reserved for the use of implementations or users.
```